osu.Game/Overlays/MusicController.cs

```diff
@@ -256,8 +257,10 @@ private PreviousTrackResult prev(bool allowProtectedTracks)
                 playableSet = getNextRandom(-1, allowProtectedTracks);
             else
             {
-                playableSet = getBeatmapSets().TakeWhile(i => !i.Value.Equals(current?.BeatmapSetInfo)).LastOrDefault(s => !s.Value.Protected || allowProtectedTracks)
-                              ?? getBeatmapSets().LastOrDefault(s => !s.Value.Protected || allowProtectedTracks);
+                var beatmapSets = getBeatmapSets(allowProtectedTracks);
+
+                playableSet = beatmapSets.TakeWhile(i => !i.Value.Equals(current?.BeatmapSetInfo)).LastOrDefault()
+                              ?? beatmapSets.LastOrDefault();
```

Comment:
multiple enumeration here (not sure how relevant, but I figure I should at least bring it up) - at the minimum I'd suggest multiple `getBeatmapSets()` calls as previously